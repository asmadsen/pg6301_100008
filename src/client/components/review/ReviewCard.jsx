import React from 'react'
import PropTypes from 'prop-types'
import CardContent from '@material-ui/core/CardContent'
import Box from '@material-ui/core/Box'
import { CardActions, Grid, Icon, makeStyles, Typography } from '@material-ui/core'
import Rating from '@material-ui/lab/Rating'
import ReadMore from '@crossfield/react-read-more'
import Button from '@material-ui/core/Button'
import Card from '@material-ui/core/Card'

const useStyles = makeStyles({
	reviewCard: {
		marginTop: 15
	}
})

const ReviewCard = ({ review, isLoggedInUser, onEdit }) => {
	const classes = useStyles()
	return (
		<Card className={classes.reviewCard}>
			{ isLoggedInUser && (
				<CardActions>
					<Box flexGrow={1} />
					<Button data-test-id="editBtn" onClick={onEdit} variant="contained">Edit</Button>
				</CardActions>
			)}
			<CardContent>
				<Box my={1}>
					<Grid container justify="space-between">
						<Typography variant="h6" component="span">
							{ review.user.alias }
						</Typography>
						<Rating
							value={review.rating}
							readOnly
						/>
					</Grid>
				</Box>
				<ReadMore
					initialHeight={350}
					readMore={props => (
						/* eslint-disable react/prop-types */
						<Button variant="text" onClick={props.onClick}>
							{props.open ? 'Read Less' : 'Read More'}
							<Icon
								style={{
									transform: `rotate( ${props.open ? '-90deg' : '90deg'})`,
									transition: 'transform 0.25s',
								}}
								className="fas fa-caret-right"
							/>
						</Button>)
						/* eslint-enable react/prop-types */
					}
				>
					<Typography component="p">{review.review}</Typography>
				</ReadMore>
			</CardContent>
		</Card>
	)
}

ReviewCard.propTypes = {
	review: PropTypes.object,
	isLoggedInUser: PropTypes.bool,
	onEdit: PropTypes.func
}

export default ReviewCard
