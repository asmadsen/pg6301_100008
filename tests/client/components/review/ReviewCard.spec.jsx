import React from 'react'
import ReviewCard from '../../../../src/client/components/review/ReviewCard'
import { shallow } from 'enzyme'

describe('client/components/review', () => {
	it('should match snapshot', () => {
		const wrapper = shallow(<ReviewCard
			review={{
				user: {
					alias: 'Alias'
				},
				rating: 4,
				// eslint-disable-next-line max-len
				review: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'
			}}
		/>)

		expect(wrapper).toMatchSnapshot()
	})

	it('should show edit button if isLoggedInUser is true', () => {
		const wrapper = shallow(<ReviewCard
			isLoggedInUser={true}
			review={{
				user: {
					alias: 'Alias'
				},
				rating: 4,
				// eslint-disable-next-line max-len
				review: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'
			}}
		/>)

		expect(wrapper.findByTestId('editBtn').exists()).toEqual(true)

		wrapper.setProps({ isLoggedInUser: false })
		expect(wrapper.findByTestId('editBtn').exists()).toEqual(false)
	})
})
