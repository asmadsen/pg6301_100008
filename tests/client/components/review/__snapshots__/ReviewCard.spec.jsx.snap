// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client/components/review should match snapshot 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-reviewCard-1"
>
  <WithStyles(ForwardRef(CardContent))>
    <Styled(MuiBox)
      my={1}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        justify="space-between"
      >
        <WithStyles(ForwardRef(Typography))
          component="span"
          variant="h6"
        >
          Alias
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Rating))
          readOnly={true}
          value={4}
        />
      </WithStyles(ForwardRef(Grid))>
    </Styled(MuiBox)>
    <ReadMore
      initialHeight={350}
      overhangSize={160}
      readMore={[Function]}
    >
      <WithStyles(ForwardRef(Typography))
        component="p"
      >
        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
      </WithStyles(ForwardRef(Typography))>
    </ReadMore>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
